src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -360,19 +361,48 @@ public void open(Configuration parameters) throws Exception {
 
     @Override
     public void close() throws Exception {
+        Throwable ex = null;
         if (eventStreamClientFactory != null) {
-            log.info("Closing Pravega eventStreamClientFactory");
-            eventStreamClientFactory.close();
+            try {
+                log.info("Closing Pravega eventStreamClientFactory");
+                eventStreamClientFactory.close();
+            } catch (Throwable e) {
+                if (e instanceof InterruptedException) {
+                    log.warn("Interrupted while waiting for eventStreamClientFactory to close, retrying ...");
+                    eventStreamClientFactory.close();
+                } else {
+                    ex = ExceptionUtils.firstOrSuppressed(e, ex);
```

Comment:
I changed to use Flink suggested way to expose exceptions in close method.
ref: https://github.com/apache/flink/blob/4467277dcf80d18b2898d4ec93c634dc33d25b2d/flink-core/src/main/java/org/apache/flink/util/ExceptionUtils.java#L253